### PR TITLE
Migrate to Castor

### PR DESCRIPTION
**Pull request by pyrech**

### CODE REVIEW

File: .castor/qa.php [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+use Castor\Attribute\AsTask;
+
+use function Castor\get_context;
+use function Castor\io;
+use function Castor\notify;
+use function Castor\parallel;
+
+use Symfony\Component\Process\Process;
+
+#[AsTask(namespace: 'qa', description: 'Run PHPUnit tests and PHP CS Fixer and PHPStan analysis')]
```
Comment (lyrixx) [inline .castor/qa.php:24]: phpunit is not run in this function

File: .castor/qa.php [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    }
+
+    return 0;
+}
+
+#[AsTask(namespace: 'qa', description: 'Run PHP CS Fixer', aliases: ['cs'])]
+function phpcsfixer(
+    #[AsOption(shortcut: 'q', description: 'Run quietly the command')]
+    bool $quiet = false,
+    #[AsOption(shortcut: 'e', description: 'Return exit code instead of process')]
```
Comment (lyrixx) [inline .castor/qa.php:66]: It does not make sens to expose this option to the CLI

File: castor.php [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    }
+}
+
+#[AsTask(description: 'Clear the application cache', namespace: 'app')]
+function cache_clear(): void
+{
+    docker_compose_run('rm -rf var/cache/ && bin/console cache:warmup');
+}
+
+#[AsTask(description: 'Migrates database schema', namespace: 'app:db')]
```
Comment (lyrixx) [inline castor.php:79]: Let's remove this one?

File: infrastructure/docker/services/php/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,109 @@
+FROM debian:11.0-slim as php-base
```
Comment (lyrixx) [inline infrastructure/docker/services/php/Dockerfile:1]: 11.9?

**Review by lyrixx — COMMENTED** [review]
Nice 👏🏼

File: .castor/infra.php [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+        '--restart=unless-stopped',
+        ...$workers,
+    ], quiet: true);
+
+    run([
+        'docker',
+        'start',
+        ...$workers,
+    ], quiet: true);
+}
```
Comment (damienalexandre) [inline .castor/infra.php:176]: **💭 thought:** ‏There are no worker on this project 😬

Comment (pyrech) [reply .castor/infra.php:176]: You really want to remove them from the infrastructure? That would complexify future updates of the stack without real gain IMO

Comment (lyrixx) [reply .castor/infra.php:176]: I agree with @damienalexandre here. If we don't use it, it does not worth keeping it

**Comment by pyrech** [comment]
PR updated

**Comment by pyrech** [comment]
Thanks for your reviews. I will merge it as is bu do not hesitate to leave another reviews if some things need to be changed

**Pull request merged**